apps/src/code-studio/components/aiTutor/aiTutorPanel.jsx

```diff
@@ -1,29 +1,94 @@
-import React from 'react';
+import React, {useState} from 'react';
 import PropTypes from 'prop-types';
+import {useDispatch} from 'react-redux';
 import i18n from '@cdo/locale';
 import style from './ai-tutor.module.scss';
+import Checkbox from '@cdo/apps/componentLibrary/checkbox';
 import AITutorPanelContainer from '@cdo/apps/code-studio/components/aiTutor/aiTutorPanelContainer';
-import CompilationAssistant from './compilationAssistant';
+import CompilationTutor from './compilationTutor';
+import ValidationTutor from './validationTutor';
+import GeneralChatTutor from './generalChatTutor';
+import {addAIResponse} from '@cdo/apps/aiTutor/redux/aiTutorRedux';
 const icon = require('@cdo/static/ai-bot.png');
 
-export default class AITutorPanel extends React.Component {
-  static propTypes = {
-    level: PropTypes.shape({
-      id: PropTypes.number.isRequired,
-      type: PropTypes.string.isRequired,
-    }),
-  };
+const AITutorPanel = props => {
+  const dispatch = useDispatch();
+  const {level} = props;
+  const isCodingLevel = level.type === 'Javalab';
+
+  const initialCheckboxes = [
+    {
+      key: 'compilation',
+      label: 'Code compilation',
+      checked: false,
+      hidden: !isCodingLevel,
+    },
+    {
+      key: 'validation',
+      label: 'Failing tests',
+      checked: false,
+      hidden: !level.hasValidation,
+    },
+    {
+      key: 'question',
+      label: 'I have a question',
+      checked: false,
+      hidden: false,
+    },
+  ];
 
-  render() {
-    const {level} = this.props;
-    const isCodingLevel = level.type === 'Javalab';
+  const [checkboxes, updateCheckboxes] = useState(initialCheckboxes);
 
-    return (
-      <AITutorPanelContainer level={level}>
-        <h3 id="ai_tutor_panel">AI Tutor</h3>
-        <img alt={i18n.aiBot()} src={icon} className={style.aiBotImg} />
-        {isCodingLevel && <CompilationAssistant levelId={level.id} />}
-      </AITutorPanelContainer>
+  const handleCheckboxChange = key => {
```

Comment:
If only one can be checked, could you store the key associated with the checked box instead of the whole array of checkbox objects and their properties on state?

Reply:
🤔 Every time a box is checked, I need to uncheck any previously checked boxes so I think I'd need state for the box to check as well as the ones to uncheck. It seemed simpler in my mind to just store all of them in state. What am I missing?

Reply:
I struggled to enroll in the experiment locally, so I couldn't actually test this, but I think something like this would do it: https://github.com/code-dot-org/code-dot-org/pull/55082/files 

Though, to Molly's point below, it looks like there's a `RadioButton` component available, which probably better represents the intended mutual exclusivity: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/componentLibrary/radioButton/README.md